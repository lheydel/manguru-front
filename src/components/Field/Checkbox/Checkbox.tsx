import MuiCheckbox, { CheckboxProps as MuiCheckboxProps } from '@material-ui/core/Checkbox';
import { FormikProps } from 'formik';
import React from 'react';

export interface CheckboxProps {
  name: string;
  values: FormikProps<any>['values'];
  handleChange: FormikProps<any>['handleChange'];
}

/**
 * Combine Material-UI Checkbox design with Formik functionnalities
 */
export const Checkbox: React.FC<CheckboxProps & MuiCheckboxProps> = ({
  name,
  values,
  handleChange,
  ...props
}) => (
  <MuiCheckbox
    name={name}
    value={values[name]}
    onChange={handleChange}
    {...props}
  />
);
