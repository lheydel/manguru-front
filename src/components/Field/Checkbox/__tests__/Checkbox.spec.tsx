import React from 'react';
import { CheckboxProps, Checkbox } from '../Checkbox';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Checkbox component', () => {
  const props: CheckboxProps = {
    name: 'blblbl',
    values: {},
    handleChange: jest.fn(),
  };

  it('should render itself and children', () => {
    const wrapper = shallow(<Checkbox {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should trigger handleChange', () => {
    const wrapper = shallow(<Checkbox {...props} />);
    wrapper.simulate('change');
    expect(props.handleChange).toHaveBeenCalled();
  });
});
