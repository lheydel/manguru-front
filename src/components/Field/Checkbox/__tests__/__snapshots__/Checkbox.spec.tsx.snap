// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render itself and children 1`] = `
<WithStyles(ForwardRef(Checkbox))
  name="blblbl"
  onChange={[MockFunction]}
/>
`;
